engine/schema/src/main/java/org/apache/cloudstack/storage/datastore/db/ImageStoreDao.java

```diff
@@ -31,6 +31,8 @@
 
     List<ImageStoreVO> findByScope(ZoneScope scope);
 
+    List<ImageStoreVO> findByScopeExcludingReadOnly(ZoneScope scope);
```

Comment:
this method name is confusing: `findByScope` but than the scope is allready set to `ZoneScope`? Wouldn't `findByZone` be better?

Also the method above could probably just be extended with a flag for `excludeReadOnly`.